src/client/bridge/gateway/mod.rs

```diff
@@ -69,7 +69,7 @@ use crate::gateway::ConnectionStage;
 /// A message either for a [`ShardManager`] or a [`ShardRunner`].
 // Once we can use `Box` as part of a pattern, we will reconsider boxing.
 #[allow(clippy::large_enum_variant)]
-#[derive(Clone, Debug)]
+#[derive(Debug)]
```

Comment:
Why have you removed it?

Reply:
This allows `EventFilter` and `event_collector::FilterOptions` to be `!Clone`. It turned out that these types that I removed the `Clone` derive from were never actually cloned, only moved. Removing the `Clone` bound allows the collector's filter closure to be `FnMut` without needing a lock, so that the closure can capture mutable data. Contrast with the existing collectors' respective `FilterOptions` types which use `Arc<dyn Fn<...>>`.

If it turns out that these types ever need to be cloned in the future `FilterOptions` can be updated from `Box<dyn FnMut<...>>` to `Arc<Mutex<FnMut<...>>>`, but for now that's not needed.